doc/error-catalog.md

```diff
@@ -169,7 +169,21 @@ They're numbered starting from `200`.
 
 ### OILS-ERR-200
 
-Example TODO
+<!--
+Generated with:
+test/runtime-errors.sh test-external_cmd_typed_args
+-->
+
+```
+  cat ("myfile")
+      ^
+[ -c flag ]:1: fatal: Unexpected typed args passed to external command 'cat' (OILS-ERR-200)
+```
+
+- External programs cannot accept [typed
```

Comment:
I might also rephrase this to compare / contrast

- Builtin commands and user-defined procs may accept [typed args], but external commands never do.